draggablepanel/src/main/java/com/github/pedrovgs/DraggableView.java

```diff
@@ -257,24 +279,48 @@ public boolean isClosed() {
    * @return true if the touch event is realized over the drag or second view.
    */
   @Override public boolean onTouchEvent(MotionEvent ev) {
+
     viewDragHelper.processTouchEvent(ev);
     if (isClosed()) {
       return false;
     }
     boolean isDragViewHit = isViewHit(dragView, (int) ev.getX(), (int) ev.getY());
     boolean isSecondViewHit = isViewHit(secondView, (int) ev.getX(), (int) ev.getY());
+    boolean interceptTap = viewDragHelper.isViewUnder(dragView, (int) ev.getX(), (int) ev.getY());
+
     if (isMaximized()) {
       dragView.dispatchTouchEvent(ev);
     } else {
       dragView.dispatchTouchEvent(cloneMotionEventWithAction(ev, MotionEvent.ACTION_CANCEL));
     }
+
+      switch(ev.getAction())
```

Comment:
Don't you prefer to move this switch inside the `else`? and move all that code to a method?

```
else{
   dragView.dispatchTouchEvent(cloneMotionEventWithAction(ev, MotionEvent.ACTION_CANCEL));
   analyzeTouchToMaximizeIfNeeded();
}
```

or:

```
else{
   dragView.dispatchTouchEvent(cloneMotionEventWithAction(ev, MotionEvent.ACTION_CANCEL));
   if(shouldMaximizeOnClick(...)){
       maximize();
    }
}
```